tiling.py

```diff
@@ -398,14 +435,17 @@ def fundamental_domains(self):
     @property
     def translational_fundamental_domains(self):
         for i in range(self._diameter):
-            pos = np.array((i - self._radius)) + np.array(self._center)
-            translate = utils.translation_matrix(
-                abs(pos) * self._group.translational_subgroup_basis[1 if pos > 0 else 0])
+            vdiameter = self._diameter if self._group.flag_ceiling and self._group.flag_floor else 1
+            for j in range(vdiameter):
+                pos = np.array((i - self._radius, (j - self._radius) if vdiameter != 1 else 0)) + np.array(self._center)
+
+                translate = utils.translation_matrix(
+                    abs(pos[0]) * self._group.translational_subgroup_basis[1 if pos[0] > 0 else 0] + pos[1] * self._group.translational_subgroup_basis[2])
 
-            prism = shapes.Prism(self._group.height, *self._group.translational_fd_vertices)
+                prism = shapes.Prism(self._group.height*2, *self._group.translational_fd_vertices)
```

Comment:
Why the `*2`? What if we don't have floor or ceiling mirrors?

Reply:
you're right, let me fix that.